inc/form.class.php

```diff
@@ -336,12 +336,7 @@ public function showForm($ID, $options=array()) {
 
       // Si le formulaire est récursif, on authorise les validateurs des sous-entités
       // Sinon uniquement les validateurs de l'entité du formulaire
-      if ($this->isRecursive()) {
-         $entites = getSonsOf('glpi_entities', $this->getEntityID());
-      } else {
-         $entites = $this->getEntityID();
-      }
-      $subentities = getEntitiesRestrictRequest("", 'pu', "", $entites, true, true);
+      $subentities = getEntitiesRestrictRequest("", 'pu', "", $this->getEntityID(), $this->isRecursive() == 1, true);
```

Comment:
ok for replacement

But why use ``== 1``, seems superflu

Reply:
seems ok to me also. Same comment as @orthagh

Reply:
== 1 has two purposes : cast into a boolean, and also ensure true is returned only when the column is_recursive contains 1.
I think ```$this->isRecursive() != 0``` is even better (this is an habit I adopted in Flyve MDM plugin for example). I avoid use of empty() to keep things simple.

Reply:
you do exactly the same thing as php core.
If you want to explicitly cast (i still think it's useless here), do it but with a real cast syntax.